xbmc/cores/AudioEngine/Engines/ActiveAE/AudioDSPAddons/ActiveAEDSP.cpp

```diff
@@ -148,17 +143,28 @@ void CActiveAEDSP::TriggerModeUpdate(bool bAsync /* = true */)
     m_databaseDSP.GetModes(m_modes[i], i);
   }
 
-  /*
-   * if any dsp processing is active restart playback
-   */
+  if (m_addonToDestroy.size() > 0)
+  {
+    for (std::list<AE_DSP_ADDON>::iterator iter = m_addonToDestroy.begin(); iter != m_addonToDestroy.end(); ++iter)
```

Comment:
should be handled in the destructor of AE_DSP_ADDON so m_addonToDestroy.clear should be enough